src/java/org/apache/cassandra/schema/SchemaConstants.java

```diff
@@ -126,4 +129,23 @@ public static Set<String> getSystemKeyspaces()
     {
         return Sets.union(Sets.union(LOCAL_SYSTEM_KEYSPACE_NAMES, REPLICATED_SYSTEM_KEYSPACE_NAMES), VIRTUAL_SYSTEM_KEYSPACE_NAMES);
     }
+
+    /**
+     * Returns the set of local and replicated system keyspace names
+     * @return all local and replicated system keyspace names
+     */
+    public static Set<String> getLocalAndReplicatedSystemKeyspaceNames()
+    {
+        return Sets.union(LOCAL_SYSTEM_KEYSPACE_NAMES, REPLICATED_SYSTEM_KEYSPACE_NAMES);
+    }
+    
+    /**
+     * Returns the set of all local and replicated system table names
+     * @return all local and replicated system table names
+     */
+    public static Set<String> getLocalAndReplicatedSystemTableNames()
+    {
+        return Sets.union(Sets.union(SystemKeyspace.TABLE_NAMES, ImmutableSet.copyOf(SchemaKeyspaceTables.ALL)),
```

Comment:
Nit: The four nested usages of `Sets.union` aren't probably the easier thing to read. Perhaps we could use instead something like:
```java
return ImmutableSet.<String>builder()
                   .addAll(SystemKeyspace.TABLE_NAMES)
                   .addAll(SchemaKeyspaceTables.ALL)
                   .addAll(TraceKeyspace.TABLE_NAMES)
                   .addAll(AuthKeyspace.TABLE_NAMES)
                   .addAll(SystemDistributedKeyspace.TABLE_NAMES)
                   .build();
```